packages/@n8n/permissions/src/hasScope.ts

```diff
@@ -7,36 +7,18 @@ export interface HasScopeOptions {
 
 export function hasScope(
 	scope: Scope | Scope[],
-	userScopes: ScopeLevels,
-	options?: HasScopeOptions,
-): boolean;
-export function hasScope(
-	scope: Scope | Scope[],
-	userScopes: Pick<ScopeLevels, 'global'>,
-	options?: HasScopeOptions,
-): boolean;
-export function hasScope(
-	scope: Scope | Scope[],
-	userScopes: Omit<ScopeLevels, 'resource'>,
-	options?: HasScopeOptions,
-): boolean;
-export function hasScope(
-	scope: Scope | Scope[],
-	userScopes: Pick<ScopeLevels, 'global'> & Partial<ScopeLevels>,
+	userScopes: Partial<ScopeLevels>,
```

Comment:
This was done so that to use the resource scope you'd need to provide the global and project scopes as well. If you want to use the project scope you need to provide the global scope as well. This way there's no way to accidentally only check just the resource level scope or something similar. I tried to do it in a more succinct type but typescript wasn't liking it :shrug:.

Reply:
Ah yes, sorry, I totally overlooked that and great that you thought of this.
After all I didn't change much on the point but let me know if you're fine with it